lib/strategy.js

```diff
@@ -75,18 +94,26 @@ function OAuthStrategy(options, verify) {
     options = undefined;
   }
   options = options || {};
-  
+
   if (!verify) { throw new TypeError('OAuthStrategy requires a verify callback'); }
   if (!options.requestTokenURL) { throw new TypeError('OAuthStrategy requires a requestTokenURL option'); }
   if (!options.accessTokenURL) { throw new TypeError('OAuthStrategy requires a accessTokenURL option'); }
-  if (!options.userAuthorizationURL) { throw new TypeError('OAuthStrategy requires a userAuthorizationURL option'); }
   if (!options.consumerKey) { throw new TypeError('OAuthStrategy requires a consumerKey option'); }
   if (options.consumerSecret === undefined) { throw new TypeError('OAuthStrategy requires a consumerSecret option'); }
-  
+
+  var provider = options.userAuthorizationURLProvider;
+  if (provider === undefined || typeof provider != 'function') {
```

Comment:
all you need is if (typeof provider !== 'function') drop *provider === undefined ||* part